odoo/fields.py

```diff
@@ -2376,6 +2376,36 @@ def __init__(self, comodel_name=Default, relation=Default, column1=Default,
             **kwargs
         )
 
+    def _get_mro_has_field_relation(self, model, name):
+        for mro in type(model).__mro__:
+            # Check that mro class is an Odoo Model
+            if not hasattr(mro, '_abstract'):
+                continue
+            # As this model is present in mro, we don't test it
+            if not mro._name or mro._name == model._name:
+                continue
+            # Field name was found in mro class
+            if name in mro.__dict__:
+                field = mro.__dict__[name]
+                # We don't know if field type was something else
+                if isinstance(field, Many2many):
+                    return self.relation == field.args.get('relation')
+        return False
+
+    def _check_relation(self, model, name):
+        if self._get_mro_has_field_relation(model, name):
+            _logger.debug(
+                _('The relation property of field %s (%s) was not defined or '
+                  'has the same value as in one of its inherited model. '
+                  'It will be redefined.') % (name, model))
+            self.relation = False
```

Comment:
I should note for our readers that this hack doesn't solve the problem in v9 because https://github.com/odoo/odoo/blob/07c111fa7255b28e901c21f2f9967bda0422ab8a/odoo/fields.py#L2379 was improved in v10. Is in v10 and upper where this hack makes sense.

Reply:
Setting `self.relation` to `False` is not acceptable on a stable version.
This is because it implies a change in the SQL schema: the field no longer uses another field's table, but now has its own table.
It will make an existing instance crash instead of returning somewhat weird results.